python/variable.cpp

```diff
@@ -126,11 +126,23 @@ Variable doMakeVariable(const std::vector<Dim> &labels, py::array &values,
                                  values.cast<std::vector<std::string>>(), unit);
     }
 
-    if (dtypeTag == core::dtype<Eigen::Vector3d>) {
+    if (dtypeTag == core::dtype<Eigen::Vector3d> ||
+        dtypeTag == core::dtype<Eigen::Quaterniond>) {
       std::vector<scipp::index> shape(values.shape(),
                                       values.shape() + values.ndim() - 1);
-      return init_1D_no_variance(
-          labels, shape, values.cast<std::vector<Eigen::Vector3d>>(), unit);
+      if (dtypeTag == core::dtype<Eigen::Vector3d>)
+        return init_1D_no_variance(
+            labels, shape, values.cast<std::vector<Eigen::Vector3d>>(), unit);
+      else {
+        std::vector<Eigen::Quaterniond> qvec;
+        auto arr = values.mutable_unchecked<double, 2>();
```

Comment:
```suggestion
        auto arr = values.unchecked<double, 2>();
```
?

Reply:
I was getting some weird behaviour with `unchecked` where values i was accessing from it were undefined (probably some object not being kept alive). I am now casting to a vector of doubles. This is probably making a copy, but it works for now, and i'd like to get this out of the way. We can optimize later if needed.